### PR TITLE
Update DiskFree to return block_size (for sosreport)

### PR DESCRIPTION
**Pull request by xiangce**
- Add an interface to return the block size to compatible with sos archives
- Remove the deprecated `shared` in examples
- Add test and refine the Examples
- Remove the dependant on defaultdict
- Fix: change the return value of two properties to list
- fix: #2502 

Signed-off-by: Xiangce Liu <xiangceliu@redhat.com>

### CODE REVIEW

**Comment by bfahr** [comment]
test this please

File: insights/parsers/df.py [hunk 1]
```diff
@@ -289,5 +313,24 @@ class DiskFree_AL(DiskFree):
                 Available      available
                 Use%           capacity
                 Mounted on     mounted_on
+        block_size (str): The unit of display values, '1K' by default.
```
Comment (JoySnow) [inline insights/parsers/df.py]: @xiangce 
As sosreport is using `df_al`, we can check `block_size` in rules to workaround this.       
For this point, do we consider to provide an `int` value of `block_size` at same time?  So we do the B/K/M/G converting here instead in each rule that use this parser.   
Like: 
```suggestion
        block_size_raw (str): The unit of display values, '1K' by default.
        block_size (int): The calculatable value of block_size_raw in byte, 1024 for '1K', 1048576 for '1M'.
```

**Comment by bfahr** [comment]
@xiangce given that we have cancelled the spec change to you want to keep any of these changes, or close this PR?

**Comment by xiangce** [comment]
> @xiangce given that we have cancelled the spec change to you want to keep any of these changes, or close this PR?

@bfahr, I'd like to keep the changes and will update for sosreport, since the command collected in sosreport archives is impacted.

File: insights/parsers/df.py (hunk 1 above)
Comment (xiangce) [reply insights/parsers/df.py]: @JoySnow - Updates applied, please have a look.

**Pull request merged**